main.nf

```diff
@@ -914,17 +1055,309 @@ else {
+		  all_contexts = params.comprehensive ? 'c, cg, ch, hcg, gch' : 'cg'
+		  """
+			biscuit vcf2bed -k $min_depth -t $all_contexts  "${vcf[0]}" > "${name}.bedgraph"   
+			
+		  """
+		}
+		/***************
+		*EXPERIMENTAL!!*
+		***************/
+	if (params.soloWCGW_file) {
+		process intersect_soloWCGW {
+			tag "$name"
+		  publishDir "${params.outdir}/methylation_extract", mode: 'copy'
+	
+			 
+		  input:
+		  set val(name), file(bedgraph) from ch_bedgraph_for_intersect_soloWCGW
+		  file soloWGCW from ch_soloWCGW_for_biscuitVCF.collect()
+
+		  output:
+		  file "*bedgraph" 
+		  script:
+		  """
+		   bedtools intersect -wa -a "${bedgraph[0].baseName}.bedgraph"  -b $soloWGCW > ${name}_soloWCGW.bedgraph 
+		  """
+		}
+	}	
+	
+    
+	if (params.epiread) {
+		/***************************
+		 THE PROCESS IS IN PROGRESS!
+		****************************/
+		process get_SNP_file { 
+			tag "$name"
+		  publishDir "${params.outdir}", mode: 'copy',
+			 saveAs: {filename ->
+					if(params.save_snp_file &&  filename != "where_are_my_files.txt") "epireads/snp/$filename"
+					else null
+			 }
+			 
+		  input:
+			set val(name), file(vcf) from ch_vcf_for_epiread
+
+		  output:
+		  file "${name}.snp.bed" into ch_snp_for_epiread
+		  script:
+		   """
+			biscuit vcf2bed -t snp "${vcf[0]}" > "${name}.snp.bed"
+		  """
+		}
+		
+		process epiread_convertion {
+			tag "$name"
+			publishDir "${params.outdir}/epireads", mode: 'copy'
+ 
+		  input:
+			set val(name), file(bam) from ch_bam_sorted_for_epiread
+			file bam_index from ch_bam_index_for_epiread
+			file fasta from ch_fasta_for_epiread.collect()
+			file fasta_index from ch_fasta_index_for_epiread.collect()
+			file snp from ch_snp_for_epiread
+			
+		  output:
+			file "*epiread" 
+
+		  script:
+		  snp_file = (snp.size()>0) ? "-B " + snp.toString() : ''
+		  if (params.single_end) {
+		  """
+		  biscuit epiread -q ${task.cpus} $fasta $bam $snp_file -o ${name}.epiread 
+		  """
+		  } else {
+			"""
+			biscuit epiread -q ${task.cpus} $fasta $bam $snp_file | sort --parallel=${task.cpus} -T .  -k2,2 -k3,3n | awk 'BEGIN{qname="";rec=""} qname==\$2{print rec"\t"\$5"\t"\$6"\t"\$7"\t"\$8;qname=""} qname!=\$2{qname=\$2;rec=\$1"\t"\$4"\t"\$5"\t"\$6"\t"\$7"\t"\$8;pair=\$3}' > ${name}.epiread 
+			"""
+		  }
+		}
+	}
+
+
+	  process biscuit_QC {
+		tag "$bam_name"
+		publishDir "${params.outdir}/biscuit_QC", mode: 'copy'
+
+		input:
+		set val(name), file(vcf) from ch_vcf_biscuit_qc
+		set val(bam_name), file(bam) from ch_bam_noDups_for_QC
+		file fasta from ch_fasta_for_biscuitQC.collect()
+		file fasta_index from ch_fasta_index_for_biscuitQC.collect()
+		file assets from ch_assets_dir_for_biscuit_qc.collect()
+		output:
+		file "*_biscuitQC" into ch_QC_results_for_multiqc
+
+		script:
+		assembly = fasta.toString().replaceAll(/\.\w+/,"")
+		"""
+		$baseDir/bin/biscuit_QC.sh -v ${vcf[0]} -o ${bam_name}.${assembly}_biscuitQC $assets $fasta ${bam_name}.${assembly} ${bam} -p ${task.cpus}
```

Comment:
I assume this script is taken from [here](https://github.com/zhou-lab/biscuit/tree/master/scripts)?

What happens in case of updates? We would need to manually copy the upstream script again.
I don't like that, it would be better to call it from the conda env for instance

Reply:
make sense. How can one do that?

Reply:
It would need to be included in the conda recipe and added to $PATH